File: Hello_World_Tutorials/tutorial-source-code/Cpp_DLL/dllist.cpp

#include "Bridges.h"
#include "DLelement.h"
#include "StudentInfo.h"

using namespace bridges;

									// helper function

DLelement&lt;StudentInfo&gt; *insertFront(DLelement&lt;StudentInfo&gt; *front,
				DLelement&lt;StudentInfo&gt; *new_el);
int main() {

	Bridges::initialize(4, "YOUR_USER_ID", "YOUR_API_KEY");

						// load student info
	int num_students = 5;
	StudentInfo students[] = {
				StudentInfo(
                      "00000000000",
                      "Gretel Chaney",
                      "CS",
                      "g.chaney@generated.com",
                      "magenta",
                      "blue",
                      9.0 
				),
				StudentInfo(
                      "00000000001",
                      "Karol Soderman",
                      "SIS",
                      "k.soderman@generated.com",
                      "magenta",
                      "red",
                      11.0
				),
				StudentInfo(
                      "00000000002",
                      "Lamont Kyler",
                      "BIO",
                      "l.kyler@generated.com",
                      "purple",
                      "green",
                      12.0
				),
				StudentInfo(
                      "00000000003",
                      "Gladys Serino",
                      "CS","g.serino@generated.com",
                      "blue",
                      "black",
                      9.0
				),
				StudentInfo("00000000004",
                      "Starr Mcginn",
                      "CS",
                      "s.mcginn@generated.com",
                      "red",
                      "cyan",
                      15.0)
			};

							// insert the students in front of the list
		DLelement&lt;StudentInfo&gt; *head = nullptr;
		for(int i = 0; i < num_students; i++){
			head = insertFront(head, new DLelement&lt;StudentInfo&gt;students[i], ""));
		}
							// add visual attributes 
		DLelement&lt;StudentInfo&gt; *curr = head, *next;
		while(curr != nullptr){
			curr->setLabel(curr->getValue().getStudentLabel());
			curr->getVisualizer()->setColor(curr->getValue().getFavoriteColor());

			DLelement&lt;StudentInfo&gt; n1, n2;
			if (curr->getNext() != nullptr) {
				next = curr->getNext();
				curr->getLinkVisualizer(next)->setColor(curr->getValue().getDislikeColor());
				next->getLinkVisualizer(curr)->setColor(curr->getValue().getDislikeColor());
			}
			curr = curr->getNext();
		}

		Bridges::setDataStructure(head);
		Bridges::visualize();

		return 0;
}

DLelement&lt;StudentInfo&gt; *insertFront(DLelement&lt;StudentInfo&gt; *front,
				DLelement&lt;StudentInfo&gt; *new_el){
	if (front == nullptr)
		return new_el;

	new_el->setNext(front);
	front->setPrev(new_el);
	
	return new_el;
}

File: _tutorials/testing/python/dllist.py
from bridges.dl_element import *
from bridges.bridges import *
from StudentInfo import *
import sys


def insertFront(front, new_el):
    if front is None:
        return new_el

    new_el.set_next(front)
    front.set_prev(new_el)

    return new_el

def main():

    args = sys.argv[1:]

    # create the Bridges object, set credentials
    bridges = Bridges(YOUR_ASSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY");

    students = []

    # load student info
    students.append(Student("00000000000",
                              "Gretel Chaney",
                              "CS",
                              "g.chaney@generated.com",
                              "magenta",
                              "blue",
                              9.0))

    students.append(Student("00000000001",
                              "Karol Soderman",
                              "SIS",
                              "k.soderman@generated.com",
                              "magenta",
                              "red",
                              11.0))

    students.append(Student("00000000002",
                              "Lamont Kyler",
                              "BIO",
                              "l.kyler@generated.com",
                              "yellow",
                              "green",
                              50.0))

    students.append(Student("00000000003",
                              "Gladys Serino",
                              "CS",
                              "g.serino@generated.com",
                              "green",
                              "magenta",
                              9.0))

    students.append(Student("00000000004",
                              "Starr Mcginn",
                              "CS",
                              "s.mcginn@generated.com",
                              "red",
                              "cyan",
                              15.0))

    # insert the students in front of the list
    head = None
    for i in range(len(students)):
        head = insertFront(head, DLelement(label = "", e = students[i]))

    # add visual attributes
    curr = head
    while curr is not None:
        curr.set_label(curr.get_value().getName())
        curr.get_visualizer().set_color(curr.get_value().getDislikeColor())

        if curr.get_next()is not None:
            next = curr.get_next()
            curr.get_link_visualizer(next).set_color(curr.get_value().getDislikeColor())
            next.get_link_visualizer(curr).set_color(curr.get_value().getDislikeColor())

        curr = curr.get_next()


    # set dat structure to be visualized
    bridges.set_data_structure(head)
    # visualize the data structure
    bridges.visualize()

if __name__ == "__main__":
    main()

